### DISCUSSION TITLE
Create a Next.js App - Installation Error - Unexpected Token

### DISCUSSION THREAD

**Discussion by RickRyan26** [vercel/next.js#18032] [Help]
Hello I'd really like to learn Next.js but the create an app isn't installing =/ Please send help

Im using Xubuntu 18 on an Oracle VM.

Node v14.13.1
NPM 6.14.8

$ npx create-next-app nextjs-blog --use-npm --example "https://github.com/vercel/next-learn-starter/tree/master/learn-starter"

npx: installed 1 in 0.911s
Creating a new Next.js app in /home/rick/Desktop/Next app/nextjs-blog.

Downloading files from repo https://github.com/vercel/next-learn-starter/tree/master/learn-starter. This might take a moment.

Installing packages. This might take a couple of minutes.

npm ERR! Unexpected token 
npm ERR!  in JSON at position 20956 while parsing '{"name":"@ampproject/toolbox-optimizer",'

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/rick/.npm/_logs/2020-10-20T01_20_55_466Z-debug.log

Aborting installation.
  npm install --loglevel error has failed.

---------------------------------

npm init next-app nextjs-blog --example "https://github.com/vercel/next-learn-starter/tree/master/learn-starter"

npx: installed 1 in 0.761s
Creating a new Next.js app in /home/rick/Desktop/Next app/nextjs-blog.

Downloading files from repo https://github.com/vercel/next-learn-starter/tree/master/learn-starter. This might take a moment.

Installing packages. This might take a couple of minutes.

yarn install v1.22.5
info No lockfile found.
[1/4] Resolving packages...
warning next > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning next > mkdirp@0.5.3: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
warning next > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
warning next > webpack > watchpack > watchpack-chokidar2 > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning next > resolve-url-loader > rework > css > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning next > chokidar > braces > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning next > chokidar > braces > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
error Received malformed response from registry for "use-subscription". The registry may be down.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Aborting installation.
  yarnpkg install --cwd /home/rick/Desktop/Next app/nextjs-blog has failed.

------------------

npm init next-app nextjs-blog --example "https://github.com/vercel/next-learn-starter/tree/master/learn-starter"

npx: installed 1 in 0.761s
Creating a new Next.js app in /home/rick/Desktop/Next app/nextjs-blog.

Downloading files from repo https://github.com/vercel/next-learn-starter/tree/master/learn-starter. This might take a moment.

Installing packages. This might take a couple of minutes.

yarn install v1.22.5
info No lockfile found.
[1/4] Resolving packages...
warning next > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning next > mkdirp@0.5.3: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
warning next > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
warning next > webpack > watchpack > watchpack-chokidar2 > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning next > resolve-url-loader > rework > css > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning next > chokidar > braces > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning next > chokidar > braces > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
error Received malformed response from registry for "use-subscription". The registry may be down.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Aborting installation.
  yarnpkg install --cwd /home/rick/Desktop/Next app/nextjs-blog has failed.

rick@rick-VirtualBox:~/Desktop/Next app$ npm init next-app nextjs-blog
npx: installed 1 in 0.755s
Creating a new Next.js app in /home/rick/Desktop/Next app/nextjs-blog.

Installing react, react-dom, and next using yarn...

yarn add v1.22.5
info No lockfile found.
[1/4] Resolving packages...
warning next > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning next > mkdirp@0.5.3: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
warning next > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
warning next > resolve-url-loader > rework > css > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning next > chokidar > braces > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning next > chokidar > braces > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
error Received malformed response from registry for "native-url". The registry may be down.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
Error: Received malformed response from registry for "react-refresh". The registry may be down.
    at MessageError.ExtendableBuiltin (/usr/share/yarn/lib/cli.js:721:66)
    at new MessageError (/usr/share/yarn/lib/cli.js:750:123)
    at Function.<anonymous> (/usr/share/yarn/lib/cli.js:50508:15)
    at Generator.next (<anonymous>)
    at step (/usr/share/yarn/lib/cli.js:310:30)
    at /usr/share/yarn/lib/cli.js:328:14
    at new Promise (<anonymous>)
    at new F (/usr/share/yarn/lib/cli.js:5301:28)
    at Function.<anonymous> (/usr/share/yarn/lib/cli.js:307:12)
    at Function.findVersionInRegistryResponse (/usr/share/yarn/lib/cli.js:50550:19)
Error: Received malformed response from registry for "webpack". The registry may be down.
    at MessageError.ExtendableBuiltin (/usr/share/yarn/lib/cli.js:721:66)
    at new MessageError (/usr/share/yarn/lib/cli.js:750:123)
    at Function.<anonymous> (/usr/share/yarn/lib/cli.js:50508:15)
    at Generator.next (<anonymous>)
    at step (/usr/share/yarn/lib/cli.js:310:30)
    at /usr/share/yarn/lib/cli.js:328:14
    at new Promise (<anonymous>)
    at new F (/usr/share/yarn/lib/cli.js:5301:28)
    at Function.<anonymous> (/usr/share/yarn/lib/cli.js:307:12)
    at Function.findVersionInRegistryResponse (/usr/share/yarn/lib/cli.js:50550:19)

Aborting installation.
  yarnpkg add --exact react react-dom next --cwd /home/rick/Desktop/Next app/nextjs-blog has failed.

-----------------------

Any help will be greatly apprecieated. I'd REALLY like to get this to work.
Thanks in advance!

**Comment by jamesmosier** [comment]
Hi @RickRyan26. It looks like maybe there was just a hiccup with the yarn registry and it might've been down momentarily. Could you give it another try? 🚀

**Reply by RickRyan26** [reply]
Hi @jamesmosier, I really appreciate your response.

I just tried it again and am still recieving the below error:

npm ERR! Unexpected token
npm ERR! in JSON at position 20956 while parsing '{"name":"@ampproject/toolbox-optimizer",'

I tried uninstalling yarn, reinstalling, then running the command again but still the same error.

Luckily I found out I can still get create-next-app to load through npm but I wish I could get my yarn to work.

It's strange that "Unexpected token" doesn't have a "<" following it. From all the documentation I've read there shouldnt be an "Unexpected token BLANK". That's my only clue that the problem might be on my end.

**Comment by mikekrell** [comment]
make sure you are using the right node version. I had this problem when I forgot to set nvm back to using node 12. I was using node 8 when I got this error.

**Reply by johnnylambada** [reply]
12 no longer works, I had to update to 19
